geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/utils/BatchScan.scala

```diff
@@ -0,0 +1,86 @@
+/***********************************************************************
+* Copyright (c) 2013-2016 Commonwealth Computer Research, Inc.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Apache License, Version 2.0
+* which accompanies this distribution and is available at
+* http://www.opensource.org/licenses/apache2.0.php.
+*************************************************************************/
+
+package org.locationtech.geomesa.hbase.utils
+
+import java.util.concurrent.{ConcurrentLinkedQueue, CountDownLatch, Executors, LinkedBlockingQueue}
+
+import org.apache.hadoop.hbase.TableName
+import org.apache.hadoop.hbase.client._
+import org.locationtech.geomesa.utils.collection.CloseableIterator
+
+import scala.collection.JavaConversions._
+
+class BatchScan(connection: Connection, tableName: TableName, ranges: Seq[Scan], threads: Int, buffer: Int)
```

Comment:
I may have written this...but thinking it through. Notes/questions:
1. Does the constructor block on the outQueue.take() and if so do we want that?
2. the next() method seems to be queuing blcoking on outQueue().take instead of having hasNext block on outQueue.take. wondering if it is better to have hasNext block and queue up the next guy instead of next(). By the time we are in next we have ```n``` waiting to be returned but we block again.